server/src/test/java/org/elasticsearch/index/mapper/KeywordFieldMapperTests.java

```diff
@@ -770,4 +771,176 @@ public void testDocValuesLoadedFromStoredSynthetic() throws IOException {
+            mapping(b -> {
+                b.startObject("host.name");
+                b.field("type", "keyword");
+                b.endObject();
+            })
+        );
+
+        final KeywordFieldMapper mapper = (KeywordFieldMapper) mapperService.documentMapper().mappers().getMapper("host.name");
+        assertTrue(mapper.fieldType().hasDocValuesSparseIndex());
+    }
+
+    public void testFieldTypeDefault_StandardMode() throws IOException {
+        final MapperService mapperService = createMapperService(
+            Settings.builder().put(IndexSettings.MODE.getKey(), IndexMode.STANDARD.name()).build(),
+            mapping(b -> {
+                b.startObject("host.name");
+                b.field("type", "keyword");
+                b.endObject();
+            })
+        );
+
+        final KeywordFieldMapper mapper = (KeywordFieldMapper) mapperService.documentMapper().mappers().getMapper("host.name");
+        assertFalse(mapper.fieldType().hasDocValuesSparseIndex());
+    }
+
+    public void testFieldTypeDefault_NonMatchingFieldName() throws IOException {
+        final MapperService mapperService = createMapperService(
+            Settings.builder().put(IndexSettings.MODE.getKey(), IndexMode.LOGSDB.name()).build(),
+            mapping(b -> {
+                b.startObject("hostname");
+                b.field("type", "keyword");
+                b.endObject();
+            })
+        );
+
+        final KeywordFieldMapper mapper = (KeywordFieldMapper) mapperService.documentMapper().mappers().getMapper("hostname");
+        assertFalse(mapper.fieldType().hasDocValuesSparseIndex());
+    }
+
+    public void testFieldTypeDefault_ConfiguredIndexed() throws IOException {
+        final MapperService mapperService = createMapperService(
+            Settings.builder().put(IndexSettings.MODE.getKey(), IndexMode.LOGSDB.name()).build(),
+            mapping(b -> {
+                b.startObject("host.name");
+                b.field("type", "keyword");
+                b.field("index", true);
+                b.endObject();
+            })
+        );
+
+        final KeywordFieldMapper mapper = (KeywordFieldMapper) mapperService.documentMapper().mappers().getMapper("host.name");
+        assertFalse(mapper.fieldType().hasDocValuesSparseIndex());
+    }
+
+    public void testFieldTypeDefault_ConfiguredDocValues() throws IOException {
+        final MapperService mapperService = createMapperService(
+            Settings.builder().put(IndexSettings.MODE.getKey(), IndexMode.LOGSDB.name()).build(),
+            mapping(b -> {
+                b.startObject("host.name");
+                b.field("type", "keyword");
+                b.field("doc_values", true);
+                b.endObject();
+            })
+        );
+
+        final KeywordFieldMapper mapper = (KeywordFieldMapper) mapperService.documentMapper().mappers().getMapper("host.name");
+        assertFalse(mapper.fieldType().hasDocValuesSparseIndex());
```

Comment:
I think supporting sparse index in this case is ok?

Reply:
I interpreted this and the other test below as "if there are doc values and all other conditions hold true we would like to have the sparse index". Anyway I think we can also decide that if `index: false` we do not create neither the inverted index nor the sparse index.

Reply:
> Anyway I think we can also decide that if index: false we do not create neither the inverted index nor the sparse index.

Let's do that?